tests/integration/ha/continuous_writes.py

```diff
@@ -56,6 +57,8 @@ async def start(self, repl_on_all_nodes: bool = False, is_bulk: bool = True) ->
         # create index if custom conf needed
         if repl_on_all_nodes:
             await self._create_fully_replicated_index()
+        else:
+            await self._create_index_with_0_all_replicas()
```

Comment:
@Mehdi-Bendriss I am scaling from 3->1, I need to have the `test_index` with 0 replicas, otherwise I will always fail. Apart from that, we are aligned any other test should always use `repl_on_all_nodes`

Reply:
Btw, I can leave a comment of "Avoid this 0-all option at all costs!" on the code...

Reply:
@phvalguima Not all tests rely on an `1-all` (full) replication factor - since many have the replication factor set on first document ingestion. And for those tests, they will essentially have the `0-all` logic you added - which would be wrong. 

I suggest you replace the arg `repl_on_all_nodes: bool = False` by: `replication_mode: Literal["0-all",  "1-all", "default"] = "default"` and replace the `else` by an `elif replication_mode == "0-all"`